src/main/java/frc/trigon/robot/subsystems/swerve/SwerveConstants.java

```diff
@@ -120,6 +120,12 @@ public abstract class SwerveConstants {
         GYRO.applyConfiguration(config);
         GYRO.setSimulationYawVelocitySupplier(SIMULATION_YAW_VELOCITY_SUPPLIER);
 
-        GYRO.registerThreadedSignal(Pigeon2Signal.YAW, Pigeon2Signal.ANGULAR_VELOCITY_Z_WORLD, PoseEstimatorConstants.ODOMETRY_FREQUENCY_HERTZ);
+        GYRO.registerThreadedSignal(Pigeon2Signal.YAW, PoseEstimatorConstants.ODOMETRY_FREQUENCY_HERTZ);
+
+        try {
+            ROBOT_CONFIG = RobotConfig.fromGUISettings();
+        } catch (IOException | ParseException e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
all the PP update stuff is all over the place. rethink how this happens so everything will be in one place.  (there are things here, in robotinit, swerve function, swerve constructor, etc. make it better.)